rust/connlib/libs/client/src/file_logger.rs

```diff
@@ -0,0 +1,44 @@
+//! Connlib File Logger
+//!
+//! This module implements a file-based logger for connlib using tracing-appender.
+//!
+//! The log files are rotated hourly.
+//!
+//! Since these will be leaving the user's device, these logs should contain *only*
+//! the necessary debugging information, and **not** any sensitive information,
+//! including but not limited to:
+//! - WiFi SSIDs
+//! - Location information
+//! - Device names
+//! - Device serials
+//! - MAC addresses
```

Comment:
Any plans on how we are going to achieve that? :)

Reply:
Other than some rudimentary log filters, not at the moment. I suppose this comment was more as a reminder to myself and any future maintainers to remember not to explicitly log (at WARN and above) any of the above. Open to ideas / suggestions though!

Reply:
I think we should never store that sensitive information directly in a string but create a wrapper `Sensitive<T>` that doesn't implement neither `Debug` nor `Display` and has a method `get_sensitive`, to reduce the chances of this happening.

Or maybe just using [`secrecy`](https://docs.rs/secrecy/latest/secrecy/index.html)'s `SecretString` is even better, I was already planning on using it for tokens.